File: Kth_largest_element_in_a_stream.java
class KthLargest {
    private int k;
    private Queue<Integer> nums;

    public KthLargest(int k, int[] nums) {
        this.k = k;
        this.nums = new PriorityQueue<>();
        for (int i = 0; i < nums.length; i++) {
            this.nums.add(nums[i]);
            if (this.nums.size() > k) {
                this.nums.poll();
            }
        }        
    }
    
    public int add(int val) {
        nums.add(val);
        if (nums.size() > k) {
            nums.poll();
        }
        return nums.peek();        
    }
}

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest obj = new KthLargest(k, nums);
 * int param_1 = obj.add(val);
 */

File: Kth_largest_element_in_a_stream.js
/**
 * @param {number} k
 * @param {number[]} nums
 */
var KthLargest = function(k, nums) {
    this.k = k;    
    this.queue = new MinPriorityQueue();

    for(let i = 0; i < nums.length; i++){
        this.queue.enqueue(nums[i]);
    }

    while(this.queue.size() > k) {
        this.queue.dequeue().element;
    }
};

/** 
 * @param {number} val
 * @return {number}
 */
KthLargest.prototype.add = function(val) {
    this.queue.enqueue(val);

    if(this.queue.size() > this.k) {
        this.queue.dequeue().element;
    }

    return this.queue.front().element;    
};

/** 
 * Your KthLargest object will be instantiated and called as such:
 * var obj = new KthLargest(k, nums)
 * var param_1 = obj.add(val)
 */
